src/core/layertree/qgslayertreemodellegendnode.cpp

```diff
@@ -604,7 +619,94 @@ void QgsSymbolLegendNode::updateLabel()
   emit dataChanged();
 }
 
+QString QgsSymbolLegendNode::evaluateLabel( QgsExpressionContext context, QString label )
+{
+  if ( !mLayerNode )
+    return QString();
+
+  QgsVectorLayer *vl = qobject_cast<QgsVectorLayer *>( mLayerNode->layer() );
+
+  if ( vl && label.isEmpty() )
+  {
+    mLabel = getCurrentLabel();
+    if ( ! mLayerNode->expression().isEmpty() )
+      mLabel = evaluateLabelExpression( "[%" + mLayerNode->expression() + "%]", vl, context );
+    else if ( mLabel.contains( "[%" ) )
+      mLabel = evaluateLabelExpression( mLabel, vl, context );
+
+    emit dataChanged();
```

Comment:
@m-kuhn the crash might come from here, this signal was present in the updateLabel function, I assumed this function was useful and necessary and I included it in this function as it is analogous to updateLabel.

Should I remove this signal or try to fix the malloc error?

Reply:
On first sight it looks like a good recipe for an infinite loop to call `evaluateLabel()` in `data()` and emit a `dataChanged()` signal within `evaluateLabel()`.

I guess this needs to be removed here and emitted specifically when this needs to be re-evaluated (i.e. on expression change, name change, other changes?, ...). If possible to emit it with more specific information about model indexes and roles that changed, even better (https://doc.qt.io/qt-5/qabstractitemmodel.html#dataChanged).

Reply:
Yeah I was not sure how data, the content of the legend and the dataChanged signal truly interracted, I assumed the behaviour would not cause a problem since it was similar to updateLabel, I'll move it to a safer place in this case.

I 'm surprised to hear that this signal trigger to call all the data functions if I understood correctly.

I'll find a fix for this thanks!

Reply:
> I 'm surprised to hear that this signal trigger to call all the data functions if I understood correctly.

Pretty much, just not directly. It notifies the view(s) that data has changed and they will start to request all data again.

Reply:
Do you suggest that I also remove the update label function, the check for feature count is done in qgsLegendModel::data.